### PR TITLE
feat(admin): expose detailed consumer-group mutation outcomes

### LINKED ISSUE #3131: feat(admin): return detailed consumer-group mutation outcomes
Child of #3126; depends on the shared outcome contracts in #3130.

Implement detailed consumer-group deletion results keyed by group ID, plus consumer-group offset alteration/deletion results keyed by TopicPartition within the requested group. Preserve successes across coordinator changes and retriable failures.

Acceptance requirements inherited from #3126:
- Add capability interfaces/extensions without breaking custom IAdminClient implementations. Keep existing convenience methods and exception behavior compatible, sharing implementation where practical.
- Retain every confirmed success/failure and original broker error code/message. Distinguish unknown outcomes after transport failure or cancellation from confirmed outcomes; missing response entries never mean success.
- Retry only unresolved/retriable work when protocol semantics permit. Preserve protection against ambiguous non-idempotent retries (#1341).
- Implement matching Dekaf.Testing behavior, public API baselines, documentation with targeted-retry examples, and the parent coverage matrix.
- Add TUnit tests for validation/empty input, cancellation/timeouts, authorization, routing changes where applicable, mixed outcomes, missing responses and ambiguous transport failure. Add appropriate Kafka integration coverage.
- Supply exact-head hosted performance evidence under CLAUDE.md. Do not add work to producer/consumer per-message paths.

Preserve existing detailed APIs, including member removal, feature updates and Streams group offsets. See the [Kafka Admin result conventions](https://kafka.apache.org/43/javadoc/org/apache/kafka/clients/admin/Admin.html).

### PR DESCRIPTION
**Pull request by thomhurst**
Consumer-group mutations return an outcome for every group or TopicPartition through IDetailedConsumerGroupMutationAdminClient and IAdminClient extensions. Group deletion batches by coordinator; offset changes batch by topic and preserve offset metadata. Ambiguous sends, omitted entries and duplicate responses remain Unknown and are never automatically replayed.

Discovery, lease and version failures are isolated per group/coordinator batch so healthy siblings still run. OffsetCommit v10 excludes unmappable topics before dispatch. If multiple requested names share one topic ID, every colliding topic is now excluded, including the first request already constructed; a third name cannot reuse the invalid identity. Their partitions remain NotAttempted while unrelated topics are sent. An entirely ambiguous request is never sent.

Repeated response topic IDs now invalidate every requested partition for that topic, including results in earlier blocks and disjoint later blocks. Replacing an earlier coordinator rejection with Unknown prevents retrying an ambiguous mutation. Independent topic outcomes remain intact. One request-local dictionary reuses the existing request topic objects for mapping, duplicate detection and partition cleanup, replacing the previous two dictionaries. Each duplicated response identity is invalidated once.

Includes matching Dekaf.Testing APIs, public API entries and targeted-retry documentation. Work and allocations are per administrative operation, coordinator batch and entity. No producer/consumer per-message path or connection lifecycle changes; no loaded stress lane is required. The existing eight-case MemoryDiagnoser fixture exercises all operation kinds, including OffsetCommit v8 and v10.

Closes #3131

Current head 1f1919674034eeb88a8918a3234a2569b14410d7 is one commit on main c113d02159c1734fdd2e5c0968cccffbc0af16c2. The quota API and shared three-argument request-builder delegate remain preserved. Final validation: 195 related detailed-admin, in-memory and topic-ID mapping tests pass on each of net10.0/net8.0; all three Kafka 4.3.1 consumer-group mutation cases pass. Both core Release targets build with zero warnings/errors. All 38 performance-tooling tests and artifact/whitespace checks pass. Four new collision/duplicate-response cases fail on the preceding e174c3b4d product and pass after the fix, including the all-ambiguous no-send case and suppression of a previously retryable response.

All eight Dry and eight Short cases complete. With DOTNET_PROCESSOR_COUNT=4 and DOTNET_GCConserveMemory=9, topic-ID mutation allocation decreases from 2.58/11.12 KB to 2.33/10.48 KB per operation for 1/16 entities. Other operation allocations are unchanged. The local candidate was slower than both preceding-product A1/A2 controls for v8 and v10; unchanged one-entity deletion/group controls also slowed by roughly 50–60% during B. Timing attribution is therefore inconclusive, not a PASS. Focused hosted comparison [34570835765](https://github.com/thomhurst/Dekaf/actions/runs/34570835765/job/103172484684) of e174c3b4d against this exact head now reports PASS for all eight cases. It measures A1/B/A2 on one Ubuntu VM with 15 iterations per phase. Topic-ID allocation decreases from 2,640 to 2,384 B at one entity and 11,384 to 10,736 B at sixteen entities; all other allocations match both controls. Topic-ID medians are 676.9 ns versus 740.2/761.5 ns and 4.41 us versus 4.95/4.87 us. No timing case crosses the regression threshold. This intentionally compares two single-commit PR revisions with the same main parent. The automatic main-based checks remain separately required; no rerun is needed for this PASS.

The shared share-offset integration setup retains the read-only coordinator-readiness fix from e174c3b4d. Its earlier failure was reproduced on #3116 with UnknownTopicOrPartition for newly created valid partitions; the original older #3241 response code was not recorded and remains unavailable. Setup waits for a nonempty topic ID and successful reads before performing the original mutation once. The technical disposition and retained validation are recorded in https://github.com/thomhurst/Dekaf/pull/3241#issuecomment-5630194366. This revision does not alter that setup, production retries or timeout settings.

Exact final source, loaded unit/integration binaries, both generated benchmark workers and raw logs/exports are retained outside removable worktrees at C:/git/Dekaf-evidence/pr-3241/1f1919674034eeb88a8918a3234a2569b14410d7: 3,436 SHA-256-verified files plus 16 raw files. Binaries contain the pre-amend InformationalVersion; source-final.zip and implementation.patch identify the tested source. Preceding-product A1 and the four failing tests are retained under topic-identities-before-e174c3b4d (1,685 verified files plus six raw files). The additional A2 control has its exact generated worker, product/fixture source and raw reports under topic-identities-control-a2-e174c3b4d. Experiment inputs are retained in topic-identities-provenance.json. The focused hosted run's original exports, logs and final table are retained in topic-identities-hosted-34570835765 with 29 SHA-256-hashed files; the GitHub artifact is the published evidence store.

Current-head CI and all automatic performance classes now pass. The subsequent review cycle found no remaining code defects; its pending-performance concern is resolved by the completed automatic gate and focused eight-case hosted PASS. All five previously replied-to bot threads are resolved after that cycle. No failed workflow was manually rerun, no performance threshold changed, and no protected-metric tradeoff is approved. Earlier implementation/rebase evidence remains attributed to the revisions linked in https://github.com/thomhurst/Dekaf/pull/3241#issuecomment-5629099379 and subsequent comments.

### CODE REVIEW

**Comment by thomhurst** [comment]
Addressed the coordinator-batching review in 4382747acf17cd116912ad207a3d8906c83f9826. Group deletion now uses one deadline and one DeleteGroups request per coordinator batch, including regrouping only confirmed coordinator rejections on retry. Completed and ambiguous outcomes are excluded even when another coordinator later fails discovery or transport. The new deterministic regression fails on the previous implementation (four requests instead of three) and passes after the fix; all 9,846 net10.0 unit tests, 215 focused net8.0 tests, Kafka integration and docs validation pass. The PR body records the archived evidence and pending hosted acceptance.

For the non-blocking simulator duplication note, the consumer setter remains separate from the existing Streams batch API: the Streams API additionally records Streams-group classification and uses topic-ID error semantics, while consumer mutations preserve consumer identity and name-based errors. Calling that batch-shaped API once per fault-scoped consumer entity would introduce a temporary array and result dictionary per entity. The small direct setter preserves those costs and behaviors; active-group, topic validation and stored-offset tests cover the consumer path.

**Comment by thomhurst** [comment]
Rebased onto the merged security APIs at main 898c9432b, producing ff381bf23c5da6cfa7247d8a1d22e9f8089d7b95. Resolved shared simulator conflicts by preserving both applyFault and getGroupId, with share-group preflight validation and coordinator retry classification intact. Both sets of public APIs, documentation rows and permanent benchmark mappings remain present. Consumer-group product code and fixture are unchanged.

158 focused tests pass on each framework, six Kafka integration cases pass, and the docs build, 36 tooling tests, artifact policy and whitespace checks pass. The PR body records the current 1,636-file archive and distinguishes earlier broad-suite/benchmark evidence. Fresh CI and review are pending.

**Comment by thomhurst** [comment]
Fixed all three pending partial-progress findings in eef8fc8dc05a8ae281ea65de6e026049070db7ac:

1. OffsetCommit v10 maps each topic independently. Mapping failures are recorded before dispatch, excluded from the pending send set, and preserved if a valid sibling's response is lost. An all-unmapped request is not sent. Tests cover multiple partitions on the missing topic, a valid sibling, lost response and the all-unmapped control.
2. Group coordinator discovery failures no longer abort discovery/sending for independent groups. Terminal failures stop the affected group only; transient pre-send failures retain bounded retry behavior after healthy batches are processed.
3. Coordinator lease and version-negotiation failures are isolated per batch. Acquired leases are released on negotiation failure. Healthy coordinators still run, completed groups never replay, and prior confirmed coordinator rejections remain intact when their retries cannot start.

The corrected regression fixture on ff381bf23 fails seven behavior cases and passes the all-unmapped control. Both final net10.0/net8.0 detailed-admin suites pass 139 tests, including existing cancellation, deadline, selective-retry and ambiguous-send checks. Ten Kafka 4.3.1 integration cases pass across the affected shared executor families. All 36 tooling tests, docs build, artifact policy and whitespace checks pass. The expanded MemoryDiagnoser fixture completes eight Short cases, including OffsetCommit v10; costs are administrative, not per produced/consumed message. No loaded stress run was dispatched.

Evidence is outside removable worktrees at C:/git/Dekaf-evidence/pr-3241/eef8fc8dc05a8ae281ea65de6e026049070db7ac with source, test binaries, generated benchmark workers, reports, raw logs/exports and verified SHA-256 inventories. Loaded binaries were built before the final amend and may report ff381bf23 as their informational SHA; their implementation matches this commit. The two integration commands reuse the same binary; their separate raw logs preserve both five-case results, while the latest HTML report covers the second command. The matching old-code run is retained under partial-preflight-final-red-ff381bf23. The earlier partial-preflight-red-ff381bf23 archive retains three NSubstitute setup failures and four behavior failures; it is explicitly not the seven-case regression evidence.

The shared private request-builder delegate now accepts the existing outcome dictionary. The topic/share/security callers only add an unused delegate parameter; their request behavior is unchanged. A later rebase of #3240 must carry this delegate signature into its configuration request builders. Current head is one commit on main 898c9432b. Awaiting fresh CI/review; threads are left unresolved until the subsequent cycle confirms the fixes.

File: src/Dekaf/Admin/AdminClient.DetailedConsumerGroupMutations.cs [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+                    if (!topics.TryGetValue(item.Topic, out var partitions)) topics.Add(item.Topic, partitions = []);
+                    partitions.Add(new() { PartitionIndex = item.Partition, CommittedOffset = item.Offset,
+                        CommittedLeaderEpoch = item.LeaderEpoch, CommittedMetadata = item.Metadata });
+                }
+                var requests = new List<OffsetCommitRequestTopic>(topics.Count);
+                var map = version >= OffsetCommitRequest.TopicIdVersion ? new OffsetTopicIdRequestMap(_metadataManager.Metadata, topics.Count) : null;
+                namesById = map is null ? null : new(topics.Count);
+                foreach (var topic in topics)
+                {
+                    var id = map?.AddTopic(topic.Key, "AlterConsumerGroupOffsetsDetailed") ?? Guid.Empty;
```
[the remark this answers is not in the document]
Comment (thomhurst) [reply src/Dekaf/Admin/AdminClient.DetailedConsumerGroupMutations.cs]: Fixed in eef8fc8dc05a8ae281ea65de6e026049070db7ac. Topic-ID mapping failures are recorded per topic and excluded before dispatch. Valid siblings are sent, unsent outcomes survive a sibling transport failure, and all-unmapped requests are not sent. Corrected old-code cases fail before the fix and pass afterward; the standard fixture now includes v10. Evidence: https://github.com/thomhurst/Dekaf/pull/3241#issuecomment-5629099379

File: src/Dekaf/Admin/AdminClient.DetailedConsumerGroupMutations.cs [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+                await EnsureInitializedAsync(token, operation).ConfigureAwait(false);
+                var byCoordinator = new Dictionary<int, List<string>>();
+                foreach (var group in groups)
+                {
+                    // A later discovery/lease failure must not replay groups completed by
+                    // an earlier coordinator. Only unsent work and confirmed rejections remain.
+                    if (results.TryGetValue(group, out var prior) && !AdminMutationResult.IsSafeCoordinatorRetry(prior))
+                        continue;
+                    token.ThrowIfCancellationRequested();
+                    var coordinator = await FindGroupCoordinatorAsync(group, token).ConfigureAwait(false);
```
[the remark this answers is not in the document]
Comment (thomhurst) [reply src/Dekaf/Admin/AdminClient.DetailedConsumerGroupMutations.cs]: Fixed in eef8fc8dc05a8ae281ea65de6e026049070db7ac. Discovery failures are isolated per group; healthy sibling groups still batch by coordinator and run within the operation deadline. Terminal discovery failures stop only that group, while transient pre-send failures retain bounded retry. Tests cover the denied group both before and between healthy siblings. Evidence: https://github.com/thomhurst/Dekaf/pull/3241#issuecomment-5629099379

File: src/Dekaf/Admin/AdminClient.DetailedConsumerGroupMutations.cs [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+                    pending.Add(group);
+                }
+
+                KafkaException? retryFailure = null;
+                foreach (var batch in byCoordinator)
+                {
+                    token.ThrowIfCancellationRequested();
+                    using var lease = await _connectionPool.LeaseConnectionAsync(batch.Key, token).ConfigureAwait(false);
+                    var version = _metadataManager.GetNegotiatedApiVersion(lease.Connection, ApiKey.DeleteGroups,
+                        DeleteGroupsRequest.LowestSupportedVersion, DeleteGroupsRequest.HighestSupportedVersion);
```
[the remark this answers is not in the document]
Comment (thomhurst) [reply src/Dekaf/Admin/AdminClient.DetailedConsumerGroupMutations.cs]: Fixed in eef8fc8dc05a8ae281ea65de6e026049070db7ac. Lease and version negotiation failures are handled per coordinator batch, and any acquired lease is disposed before continuing. Tests reproduce I/O, registration and unsupported-version failures on the first coordinator while asserting the later healthy group is sent exactly once. Ambiguous SendAsync failures retain Unknown semantics. Evidence: https://github.com/thomhurst/Dekaf/pull/3241#issuecomment-5629099379

**Comment by thomhurst** [comment]
Rebased as f76135147de0749a3ce11f98a1be9e9451e7d5e2 onto main c113d02159c1734fdd2e5c0968cccffbc0af16c2. Preserved the new quota APIs, routing, docs and benchmark mappings; adapted its builder to the existing three-argument callback. Consumer-group fixes and benchmark source are unchanged.

168 focused unit tests pass per framework (net10.0/net8.0), plus 38 tooling tests, docs build, artifact policy and whitespace. Integration validation is **10/11**, not all green: ShareOffsets_RetainMixedOutcomesAndDeleteWholeTopics failed its first success assertion. Its report lacks the returned error. A single focused diagnostic run passed with two successes and the expected UnknownTopicOrPartition for the missing topic. This does not establish a cause or resolve the failure. Kept outcome logging and the fixture logger; no assertion, timeout or product retry was changed.

**Do not merge solely on later green CI.** The intermittent share-offset failure needs investigation or an explicit technical/maintainer disposition. Original failure evidence: `C:/git/Dekaf-evidence/pr-3241/quota-rebase-share-failure-7505cf33f` (1,648 SHA-verified files). Final source, diagnostic run and raw rebase checks: `C:/git/Dekaf-evidence/pr-3241/quota-rebase-final`. Earlier Short measurements remain scoped to eef8fc8dc05a8ae281ea65de6e026049070db7ac; no benchmark rerun is claimed for this rebase.

**Comment by thomhurst** [comment]
The red net10.0 job in CI run 34560406089 is an artifact-finalization infrastructure failure, not a failing test: all 9,928 unit cases, 39 stress-tool cases and 30 protocol-fuzz cases passed, along with the API and binary-compatibility checks. Both artifact uploads transferred their complete content and then received `Failed to FinalizeArtifact ... (403) Forbidden: Error from intermediary`. The original full job log is retained at C:/git/Dekaf-evidence/pr-3241/ci-34560406089-unit.log.

This is attempt 1. I am using the single permitted infrastructure retry for job 103141866160 and its dependents. No source, assertion, timeout or performance tolerance changes accompany it. This retry does not resolve or authorize merging past the separately recorded intermittent ShareOffsets_RetainMixedOutcomesAndDeleteWholeTopics failure.

**Comment by thomhurst** [comment]
The first diagnostic had isolated the share-offset test, so I checked the original 11-test parallel context once with the outcome logging already present in f76135147de0749a3ce11f98a1be9e9451e7d5e2. This uses Kafka 4.3.1, net10.0, the same five AdminDetailed classes, default TUnit parallelism and the existing four-test fixture gate. No product/test source, assertion, timeout or retry policy changed.

All 11 tests passed. The share-offset operation returned two Succeeded/None outcomes and Failed/UnknownTopicOrPartition for the missing topic. This does not explain the original failure, which completed in 88.9 ms and did not retain its returned broker error. Neither the focused nor this parallel diagnostic reproduces it; no further identical run is justified and the manual merge blocker remains.

Exact current source, loaded integration binaries, HTML/JSON reports, raw output, returned outcomes and experiment settings are retained at C:/git/Dekaf-evidence/pr-3241/share-offset-parallel-f76135147. Binary and report copies were SHA-256 compared with their worktree originals. This is new diagnostic evidence, not recovery of the missing original outcome. No new fix or performance acceptance is claimed.

**Comment by thomhurst** [comment]
Technical disposition for the intermittent share-offset setup failure noted in https://github.com/thomhurst/Dekaf/pull/3241#issuecomment-5629260362: fixed in e174c3b4d8941f254854a8d9b546dca39d1ae8a3.

The same shared test failed in #3116 hosted CI and was reproduced locally in the full Admin category with outcome diagnostics. Both valid, newly created partitions returned UnknownTopicOrPartition before the success assertions. The original #3241 failure did not record its response code; that evidence has not been recovered. The new reproduction and Kafka 4.3.1 coordinator source attribution are recorded at https://github.com/thomhurst/Dekaf/pull/3116#issuecomment-5630112631.

The test now uses a read-only DescribeShareGroupOffsets preflight to wait until the single-broker share coordinator reports a nonempty topic ID and successful reads for both valid partitions. It then performs the original mutation once. The missing-topic assertions, production retry semantics and existing timeout settings are unchanged. This addresses the demonstrated setup-readiness cause rather than treating a passing repeat as attribution.

Validation on the final code: the original 11-case parallel class scope passes on net10.0/Kafka 4.3.1; both share-offset cases pass on net8.0/Kafka 4.2.1, with DOTNET_PROCESSOR_COUNT=4 and DOTNET_GCConserveMemory=9. Artifact policy and whitespace checks pass. Product and benchmark source are unchanged by this fix, so their prior validation remains scoped to the revisions in the PR body.

Exact source, 344 net10.0 and 351 net8.0 binary files, four reports, commands, raw logs and SHA-256 inventories are archived and verified at C:/git/Dekaf-evidence/pr-3241/share-readiness-final-f76135147. The binaries contain the pre-amend InformationalVersion; the final source archive identifies this commit. Fresh hosted CI and subsequent review must complete before merging.

File: src/Dekaf/Admin/AdminClient.DetailedConsumerGroupMutations.cs [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+            (_, response) =>
+            {
+                var results = new Dictionary<TopicPartition, AdminMutationResult>();
+                var current = namesById is null ? null : _metadataManager.Metadata.CaptureSnapshot();
+                foreach (var topic in response.Topics)
+                {
+                    var name = topic.Name;
+                    if (namesById is not null)
+                    {
+                        if (!namesById.TryGetValue(topic.TopicId, out name)) continue;
```
[the remark this answers is not in the document]
Comment (thomhurst) [reply src/Dekaf/Admin/AdminClient.DetailedConsumerGroupMutations.cs]: Fixed in 1f1919674034eeb88a8918a3234a2569b14410d7. A duplicate response topic ID now invalidates every requested partition for that topic, including earlier results and disjoint later blocks. The ID remains invalid so a third block cannot confirm anything. Earlier coordinator rejections become Unknown, preventing an unsafe retry; unrelated topic results remain intact.

Both new cases fail on e174c3b4d and pass after the fix: three repeated topic blocks with disjoint partitions, once with an initial success and once with an initial NotCoordinator rejection. The final 195 related unit cases pass on each framework and all three Kafka consumer-group mutation cases pass. The request-local map reuses existing request topic objects, so duplicate cleanup traverses only that topic's partitions. Current sources, measured binaries, generated benchmark workers and raw results are SHA-256 retained at C:/git/Dekaf-evidence/pr-3241/1f1919674034eeb88a8918a3234a2569b14410d7; the failing source/binaries are in topic-identities-before-e174c3b4d. Local allocations improve, but timing attribution is inconclusive and a focused hosted comparison is pending. This thread remains open for a subsequent review/CI cycle.

File: src/Dekaf/Admin/AdminClient.DetailedConsumerGroupMutations.cs [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+                foreach (var topic in topics)
+                {
+                    Guid id;
+                    try
+                    {
+                        id = map?.AddTopic(topic.Key, "AlterConsumerGroupOffsetsDetailed") ?? Guid.Empty;
+                    }
+                    catch (KafkaException exception) when (exception.ErrorCode == ErrorCode.UnknownTopicId)
+                    {
+                        (excludedTopics ??= []).Add(topic.Key);
```
[the remark this answers is not in the document]
Comment (thomhurst) [reply src/Dekaf/Admin/AdminClient.DetailedConsumerGroupMutations.cs]: Fixed in 1f1919674034eeb88a8918a3234a2569b14410d7. When requested names collide on one metadata ID, all corresponding topics are excluded before dispatch, including the first already-built request. The map keeps an invalid entry for that ID so a third colliding name cannot reuse it. All affected partitions are NotAttempted, healthy siblings still run, and an all-ambiguous call sends no mutation.

Both new metadata-collision cases fail on e174c3b4d and pass after the fix. They cover three colliding names, multiple partitions on the first topic, an interleaved healthy sibling and the all-ambiguous case. All 195 related unit cases pass on net10.0/net8.0 and all three Kafka consumer-group mutation cases pass. The implementation replaces the previous two mapping dictionaries with one dictionary that reuses existing request topic objects. Eight Dry/Short cases complete; topic-ID allocations fall from 2.58/11.12 KB to 2.33/10.48 KB at 1/16 entities. Local timing remains inconclusive because unchanged controls also slowed, so no timing PASS or tradeoff is claimed. Exact evidence is retained at the paths in the updated PR body. Fresh hosted validation and subsequent review are required before resolving this thread.

**Comment by thomhurst** [comment]
1f1919674034eeb88a8918a3234a2569b14410d7 fixes both reviewed OffsetCommit v10 identity defects: all requested names sharing a metadata ID are excluded before send, and repeated response topic blocks invalidate the whole topic without unsafe retries. Four new cases fail on e174c3b4d and pass after the fix. Final validation passes 195 related unit cases per framework, all three Kafka 4.3.1 mutation cases, both core builds and 38 tooling checks. Exact source, binaries, generated workers and raw evidence are archived with verified SHA-256 inventories as recorded in the PR body.

Eight Dry and eight Short cases complete. Topic-ID allocation falls from 2.58/11.12 KB to 2.33/10.48 KB for 1/16 entities; other operation allocations stay unchanged. Local timing does not establish acceptance: candidate v10 measured 797.8 ns/4.331 us versus A1 508.4 ns/3.191 us and A2 495.5 ns/3.231 us. Name-based alteration also slowed, and unchanged one-entity delete/group controls slowed by roughly 50–60% during B. These observations leave attribution inconclusive; they are not discarded or called a PASS.

One focused hosted comparison is now dispatched: https://github.com/thomhurst/Dekaf/actions/runs/34570835765. Inputs: base_sha=e174c3b4d8941f254854a8d9b546dca39d1ae8a3, head_sha=1f1919674034eeb88a8918a3234a2569b14410d7, filters=*.Unit.AdminDetailedConsumerGroupMutationBenchmarks.*. The selector confirms exactly one class, eight cases. It uses the existing Ubuntu A1/B/A2 workflow and thresholds to compare the corrected source against the preceding PR implementation. These single-commit PR revisions intentionally share parent c113d021 rather than being ancestors of each other. This is additional revision-to-revision evidence; the automatic main-based gate remains required.

Verdict: pending. No workflow was manually rerun and no tradeoff is approved. Do not merge until this comparison, current-head CI and the subsequent review cycle complete; the new review threads remain open for that cycle.

**Comment by thomhurst** [comment]
The focused hosted comparison is complete: PASS for all eight cases on exact head 1f1919674034eeb88a8918a3234a2569b14410d7 versus preceding e174c3b4d. Run/table: https://github.com/thomhurst/Dekaf/actions/runs/34570835765/job/103172484684. Inputs remain base_sha=e174c3b4d8941f254854a8d9b546dca39d1ae8a3, head_sha=1f1919674034eeb88a8918a3234a2569b14410d7, filters=*.Unit.AdminDetailedConsumerGroupMutationBenchmarks.*.

The Ubuntu job measured A1/B/A2 on one VM, with 15 iterations per phase. Topic-ID allocation falls from 2,640 to 2,384 B for one entity and 11,384 to 10,736 B for sixteen. All other allocations match both controls. Topic-ID medians are 676.9 ns versus 740.2/761.5 ns and 4.41 us versus 4.95/4.87 us. Every timing case passes the existing threshold. This resolves the specifically pending revision-to-revision performance comparison noted in the current-head review; the earlier drifting Windows measurements remain retained as diagnostics.

The original hosted exports/logs and final table are retained at C:/git/Dekaf-evidence/pr-3241/topic-identities-hosted-34570835765 with a 29-file SHA-256 inventory. No rerun was requested, no threshold changed and no protected-metric tradeoff is needed for this comparison. The PR body is updated. The automatic main-based gate still has two pending classes, so merging remains deferred until every required check and review condition is satisfied.
